File: 2243. Calculate Digit Sum of a String/solution.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
 var digitSum = function(s, k) {
  let result = "";
  while(s.length > k){
    let total = 0;
    for (let index = 0; index < s.length; index++){
      let value = s[index];
      if (index != 0 && index % k == 0){
        result += total.toString();
        total = 0;
      }
      total += parseInt(value);
    }
    result += total.toString();
    s = result;
    result = "";
  }

  return s;
};

File: 2243. Calculate Digit Sum of a String/solution.cpp
class Solution {
public:
  string digitSum(string s, int k) {
    string result = "";
    while(s.size() > k){
      int total = 0;
      for (int index = 0; index < s.size(); index++){
        char value = s[index];
        if (index != 0 && index % k == 0){
          result += to_string(total);
          total = 0;
        }
        total += (int) value - 48;
      }
      result += to_string(total);
      s = result;
      result = "";
    }

    return s;
  }
};